example(sap_ariba): add sap_ariba connector example (#310)
fivetran/fivetran_connector_sdk: connectors/sap_ariba/README.md

| `itemShipToPostalCode` | STRING | Postal code. |
| `itemShipToCountry` | STRING | Country name. |
| `isoCountryCode` | STRING | ISO 2-letter country code. |
| `itemShipToCode` | STRING | Internal ship-to code. |
| `itemLocation` | STRING | Delivery location name. |
| `requestedDeliveryDate` | UTC_DATETIME | Requested delivery date. |
| `requestedShipmentDate` | UTC_DATETIME | Requested shipment date. |
| `row_id` | INT | Sequential ID used for uniqueness. |

## Additional considerations
The examples provided are intended to help you effectively use Fivetran's Connector SDK. While we've tested the code, Fivetran cannot be held responsible for any unexpected or negative consequences that may arise from using these examples. For inquiries, please reach out to our Support team.
When adapting this connector for production:
- Replace the Sandbox API URL with your tenant’s production endpoint.
- Increase retry delay (`__RETRY_DELAY`) if you encounter rate limits.
- Confirm that your API key has permissions for both Purchase Order and Item endpoints.
- All timestamps are normalized to UTC (`UTC_DATETIME`).
- The connector is idempotent and safe for repeated runs.

> For questions or issues, contact Fivetran Support.

Reviewer: ```suggestion

```

# SAP Ariba Purchase Orders Connector Example

## Connector overview
The SAP Ariba Purchase Orders connector demonstrates how to use the Fivetran Connector SDK to extract **purchase order (PO)** and **line-item** data from the [SAP Ariba API](https://api.sap.com/api/api_purchase_orders/overview). It loads the data into a Fivetran destination for analytics and reporting.

> The connector supports **incremental syncing** by maintaining timestamps in its internal state. It fetches new or modified purchase orders and items and marks processed rows with checkpoints for reliable continuation.

Reviewer: ```suggestion
The connector supports incremental syncing by maintaining timestamps in its internal state. It fetches new or modified purchase orders and items, and marks processed rows with checkpoints for reliable continuation.
```

    - Linux: Distributions such as Ubuntu 20.04 or later, Debian 10 or later, or Amazon Linux 2 or later (arm64 or x86_64)
- SAP Ariba API key (sandbox or production)
- Outbound internet access to `sandbox.api.sap.com` (or your production Ariba host)

## Getting started
Refer to the [Connector SDK Setup Guide](https://fivetran.com/docs/connectors/connector-sdk/setup-guide) for setup instructions, including installing dependencies, configuring authentication, and running local tests.

## Features
- Fetches purchase orders and line items from the SAP Ariba API.
- Performs **incremental syncs** using internal timestamp checkpoints.
- Supports pagination with `$skip` and `$top` query parameters.
- Converts timestamps to ISO 8601 UTC format.
- Retries automatically on network failures or rate limit errors.
- Performs schema-aware upserts for idempotent loading into Fivetran destinations.

## Configuration file
The connector reads configuration settings from `configuration.json`, which defines the authentication parameters for SAP Ariba.

```
{
  "api_key": <"YOUR_SAP_ARIBA_API_KEY">
}
```

| Key | Required | Description |
|-----|-----------|-------------|

> | `APIKey` | Yes | Your SAP Ariba API key for authentication. |

Reviewer: ```suggestion
- `api_key` (required): Your SAP Ariba API key for authentication.
```
